### ISSUE TITLE
kopy.io client side encryption settings?

### ISSUE THREAD

**Issue by jellisii** [Komodo/KomodoEdit#562]
Kopy.io integration is cool, but the ability to apply a client side key would (in my mind) make the feature much nicer to use.

**Comment by dch** [comment]
yes please to this by default. It would reduce the problems from uploading something inadvertently.

**Comment by Naatan** [comment]
This is what client side encryption is for, are you saying you want this to be the default?

I don't see any value in providing your own key, especially considering kopy is designed to be as low friction as possible. Could you explain your use case? I'm not against it, I just don't want to introduce any features that will add needless layers of complexion both in code and user experience.

**Comment by dch** [comment]
TBH I don't care about providing my own key by default, but atm we use a whole bunch of tools just to share stuff - some private, some secure, some just boring and not needing the full gist.github.com treatment.

Implementing something like the client side part of https://0bin.net/ and handing me back the URL for pasting into a chat session, without needing to switch to the terminal and back again is perfect. https://github.com/zalora/zerobin is another example using the SJCL  http://bitwiseshiftleft.github.io/sjcl/

**Comment by dch** [comment]
I should clarify this:

when using kopy.io directly from Komodo (the best way!), there appears to be no way to control whether client-side encryption is used or not. There are no settings in Preferences, nor any when I right-click to choose settings, so by default all kopies (sic) directly from the IDE are public, with no apparent way to change that. Maybe I missed an option or setting somewhere?

**Comment by Naatan** [comment]
Gotcha, thanks for clarifying :)
